dbgd.c

```diff
@@ -211,21 +211,55 @@ static int dbgd_mem_read(Dbg__Request *req, Dbg__Response *res)
     res->address = req->address;
     res->has_address = 1;
 
-    res->size = 1; /* FIXME: add word, dword, qword support */
-    res->has_size = 1;
+    res->data.len  = req->size;
+    res->data.data = malloc(res->data.len);
+    res->has_data = 1;
 
-    res->value = *((uint8_t*)(req->address));
-    res->has_value = 1;
+    unsigned int i = 0;
+    unsigned int s = req->size;
+
+    while(s >= 4) {
```

Comment:
I kind of agree, but on the other hand it is probably an unrealistic use case:

I believe all GPU and APU registers are 4 byte.
With the ACI some registers are 1 or 2 but one can also use them with a 4 byte access [which makes sense as it's some kind of atomic operation then]. Even for the ACI, the sizes are kind of weirdly mixed, so it's not a block of 2 byte registers, but the register sizes of consecutive registers is more like: 4, 4, 4, 1, 2, 1, 4, 2, 4, 1, 4 [or something = not actual values]

If you don't access MMIO, then it doesn't matter at all how it's being accessed.
So really the 4 byte write loop will handle almost all MMIO cases and work for RAM too.

- I'll decide on this tomorrow.